context-propagation/src/test/java/io/micrometer/context/integration/Slf4jThreadLocalAccessorTests.java

```diff
@@ -0,0 +1,108 @@
+import org.junit.jupiter.api.Test;
+import org.slf4j.MDC;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class Slf4jThreadLocalAccessorTests {
+
+    @Test
+    void shouldCopyEntireMdcContentsToNewThread() throws InterruptedException {
+        ContextRegistry registry = new ContextRegistry().registerThreadLocalAccessor(new Slf4jThreadLocalAccessor());
+
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+
+        CountDownLatch latch = new CountDownLatch(1);
+        AtomicReference<String> value1InOtherThread = new AtomicReference<>();
+        AtomicReference<String> value2InOtherThread = new AtomicReference<>();
+
+        MDC.put("key1", "value1");
+        MDC.put("key2", "value2");
+
+        ContextSnapshot snapshot = ContextSnapshotFactory.builder()
+            .contextRegistry(registry)
+            .clearMissing(true)
+            .build()
+            .captureAll();
+
+        executorService.submit(() -> {
+            try (ContextSnapshot.Scope scope = snapshot.setThreadLocals()) {
+                value1InOtherThread.set(MDC.get("key1"));
+                value2InOtherThread.set(MDC.get("key2"));
+            }
+            latch.countDown();
+        });
+
+        latch.await(100, TimeUnit.MILLISECONDS);
+
+        assertThat(value1InOtherThread.get()).isEqualTo("value1");
+        assertThat(value2InOtherThread.get()).isEqualTo("value2");
+
+        executorService.shutdown();
+    }
+
+    @Test
+    void shouldCopySelectedMdcContentsToNewThread() throws InterruptedException {
+        ContextRegistry registry = new ContextRegistry()
+            .registerThreadLocalAccessor(new Slf4jThreadLocalAccessor("key1", "key2"));
+
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+
+        CountDownLatch latch = new CountDownLatch(1);
+        AtomicReference<String> value1InOtherThread = new AtomicReference<>();
+        AtomicReference<String> value2InOtherThread = new AtomicReference<>();
+        AtomicReference<String> value3InOtherThread = new AtomicReference<>();
+
+        MDC.put("key1", "value1");
+        MDC.put("key2", "value2");
+        MDC.put("key3", "value3");
+
+        ContextSnapshot snapshot = ContextSnapshotFactory.builder()
+            .contextRegistry(registry)
+            .clearMissing(true)
+            .build()
+            .captureAll();
+
+        executorService.submit(() -> {
+            try (ContextSnapshot.Scope scope = snapshot.setThreadLocals()) {
+                value1InOtherThread.set(MDC.get("key1"));
+                value2InOtherThread.set(MDC.get("key2"));
+            }
+            latch.countDown();
+        });
+
+        latch.await(100, TimeUnit.MILLISECONDS);
+
+        assertThat(value1InOtherThread.get()).isEqualTo("value1");
+        assertThat(value2InOtherThread.get()).isEqualTo("value2");
+        assertThat(value3InOtherThread.get()).isNull();
```

Comment:
`value3InOtherThread.set` is never called. It should be called in the scope to prove that within the scope the MDC for `key3` is not set. I've polished this in 3602a91caa82851d0baeca9226a7e9cc3b5ad28f